scs-app/src/app/fulfill_major/fulfill_major.component.ts

```diff
@@ -0,0 +1,40 @@
+import { Component, OnInit } from '@angular/core';
+import { Observable } from 'rxjs';
+import { debounceTime, distinctUntilChanged, map } from 'rxjs/operators';
+
+import { CreditCalcService } from '../services/credit-calc/credit-calc.service';
+
+@Component({
+  selector: 'fulfill-major',
+  templateUrl: './fulfill_major.component.html',
+  styleUrls: ['./fulfill_major.component.scss'],
+  providers: [ CreditCalcService ]
+})
+export class FulfillMajorComponent implements OnInit {
+
+  private courselist: {course_name: string, fulfills: string}[] = [];
+  public user_courses: {course_name: string, fulfills: string}[] = [];
+
+  constructor(private creditCalcService: CreditCalcService) { }
+
+  ngOnInit() {
+    this.courselist = this.creditCalcService.getCourselist();
+  }
+
+  search = (text$: Observable<string>) =>
```

Comment:
let's put a comment here on where we got this from like the link we c+v'ed so people can reference in the future :)